core_backend/app/dashboard/models.py

```diff
@@ -0,0 +1,256 @@
+        )
+        .join(ResponseFeedbackDB.query)
+        .where(
+            ResponseFeedbackDB.query.has(user_id=user_id),
+            ResponseFeedbackDB.feedback_datetime_utc <= start_date,
+            ResponseFeedbackDB.feedback_datetime_utc
+            > start_date - (end_date - start_date),
+        )
+        .group_by(ResponseFeedbackDB.feedback_sentiment)
+    )
+
+    feedback_prev_period = (await asession.execute(statement_previous)).fetchall()
+    feedback_prev_period_dict = {row[0]: row[1] for row in feedback_prev_period}
+
+    feedback_stats = get_feedback_stats(
+        feedback_curr_period_dict, feedback_prev_period_dict
+    )
+
+    return ResponseFeedbackStats.model_validate(feedback_stats)
+
+
+async def get_content_feedback_stats(
+    user_id: int, asession: AsyncSession, start_date: date, end_date: date
+) -> ContentFeedbackStats:
+    """
+    Retrieve statistics for content feedback
+    """
+    statement_curr = (
+        select(
+            ContentFeedbackDB.feedback_sentiment,
+            func.count(ContentFeedbackDB.feedback_id),
+        )
+        .join(ContentFeedbackDB.content)
+        .where(
+            ContentFeedbackDB.content.has(user_id=user_id),
+            ContentFeedbackDB.feedback_datetime_utc <= end_date,
+            ContentFeedbackDB.feedback_datetime_utc > start_date,
+        )
+        .group_by(ContentFeedbackDB.feedback_sentiment)
+    )
+
+    feedback_curr_period = (await asession.execute(statement_curr)).fetchall()
+    feedback_curr_period_dict = {row[0]: row[1] for row in feedback_curr_period}
+
+    statement_previous = (
+        select(
+            ContentFeedbackDB.feedback_sentiment,
+            func.count(ContentFeedbackDB.feedback_id),
+        )
+        .join(ContentFeedbackDB.content)
+        .where(
+            ContentFeedbackDB.content.has(user_id=user_id),
+            ContentFeedbackDB.feedback_datetime_utc <= start_date,
+            ContentFeedbackDB.feedback_datetime_utc
+            > start_date - (end_date - start_date),
+        )
+        .group_by(ContentFeedbackDB.feedback_sentiment)
+    )
+
+    feedback_prev_period = (await asession.execute(statement_previous)).fetchall()
+    feedback_prev_period_dict = {row[0]: row[1] for row in feedback_prev_period}
+
+    feedback_stats = get_feedback_stats(
+        feedback_curr_period_dict, feedback_prev_period_dict
+    )
+
+    return ContentFeedbackStats.model_validate(feedback_stats)
+
+
+def get_feedback_stats(
+    feedback_curr_period_dict: dict, feedback_prev_period_dict: dict
+) -> Dict[str, Union[int, float]]:
+    """
+    Get feedback statistics
+    """
+    n_positive_curr = feedback_curr_period_dict.get("positive", 0)
+    n_negative_curr = feedback_curr_period_dict.get("negative", 0)
+    n_positive_prev = feedback_prev_period_dict.get("positive", 0)
+    n_negative_prev = feedback_prev_period_dict.get("negative", 0)
+
+    if n_positive_prev == 0:
```

Comment:
I think we should have a function like:
```
def get start(curr,prev):
     if prev == 0:
        percentage_positive_increase = 0.0
    else:
        percentage_positive_increase = (
            curr - prev
        ) / prev
    return percentage_positive_increase

```
So that we can use it twice in that function and in other functions using that logic

Reply:
Good idea. I'll refactor this